lib/ui/platform_dispatcher.dart

```diff
@@ -0,0 +1,1115 @@
+  /// obscured by system UI (such as the system notification area), or physical
+  /// intrusions in the display (e.g. overscan regions on television screens or
+  /// phone sensor housings).
+  final WindowPadding viewPadding;
+
+  /// The number of physical pixels on each side of this screen rectangle into
+  /// which the application can place a view, but where the operating system
+  /// will consume input gestures for the sake of system navigation.
+  final WindowPadding systemGestureInsets;
+
+  /// The number of physical pixels on each side of this screen rectangle into
+  /// which the application can place a view, but which may be partially
+  /// obscured by system UI (such as the system notification area), or physical
+  /// intrusions in the display (e.g. overscan regions on television screens or
+  /// phone sensor housings).
+  final WindowPadding padding;
+
+  @override
+  String toString() {
+    return '$runtimeType[screenName: $screenName, geometry: $geometry, devicePixelRatio: $devicePixelRatio]';
+  }
+}
+
+/// The immutable configuration information for a view.
+///
+/// This is used by the [PlatformDispatcher] to hold a view's configuration
+/// details so that they can be atomically updated by the engine.
+///
+/// See also:
+///
+///  * [FlutterView], which represents a particular view, including this information.
+///  * [PlatformDispatcher.views], which provides access to the views on
+///    this device.
+class ViewConfiguration {
+  /// A const constructor for an immutable [ViewConfiguration].
+  const ViewConfiguration(
+    this.screen,
+  { this.window,
+    this.geometry = Rect.zero,
+    this.depth = double.maxFinite,
+    this.visible = false,
+    this.viewInsets = WindowPadding.zero,
+    this.viewPadding = WindowPadding.zero,
+    this.systemGestureInsets = WindowPadding.zero,
+    this.padding = WindowPadding.zero,
+  });
+
+  /// Copy this configuration with some fields replaced.
+  ViewConfiguration copyWith({
+    Screen? screen,
+    FlutterWindow? window,
+    Rect? geometry,
+    double? depth,
+    bool? visible,
+    WindowPadding? viewInsets,
+    WindowPadding? viewPadding,
+    WindowPadding? systemGestureInsets,
+    WindowPadding? padding,
+  }) {
+    return ViewConfiguration(
+      screen ?? this.screen,
+      window: window ?? this.window,
+      geometry: geometry ?? this.geometry,
+      depth: depth ?? this.depth,
+      visible: visible ?? this.visible,
+      viewInsets: viewInsets ?? this.viewInsets,
+      viewPadding: viewPadding ?? this.viewPadding,
+      systemGestureInsets: systemGestureInsets ?? this.systemGestureInsets,
+      padding: padding ?? this.padding,
+    );
+  }
+
+  /// The screen that this view should appear on.
+  ///
+  /// This is the screen that the upper left corner of the view appears on.
+  final Screen screen;
+
+  /// The top level view into which the view is placed and its geometry is
```

Comment:
This doc comment confused me a little but because it only talks about views and the property is named "window"...